fitbenchmarking/results_processing/tests/test_problem_summary_page.py

```diff
@@ -151,15 +152,20 @@ def setUp(self):
         """
         Setup for create function tests
         """
-        self.results_dir = TemporaryDirectory()
-        self.supp_dir = os.path.join(self.results_dir.name, 'support_pages')
-        self.fig_dir = os.path.join(self.supp_dir, 'figures')
+        with TemporaryDirectory() as directory:
+            directory_name = directory
+            self.supp_dir = os.path.join(directory_name, 'support_pages')
+            self.fig_dir = os.path.join(self.supp_dir, 'figures')
         os.makedirs(self.fig_dir)
-        results, self.options = generate_mock_results(
-            self.results_dir.name
-        )
+        results, self.options = generate_mock_results(directory_name)
         self.best_results, self.results = preprocess_data(results)
 
+    def tearDown(self) -> None:
+        """
+        Tear down the test suite.
+        """
+        shutil.rmtree(self.fig_dir)
```

Comment:
We need to delete the full path. i.e. for 
```
<place_where_tempfile_makes_dirs>/<new_dir>/support_pages/figures
```

This will just delete the `figures` directory, leaving:
```
.../<new_dir>/support_pages
```

Reply:
Good spot, its should be delete the full temporary directory now!